From 8daaa8c27794653d02854c5982669a7638473224 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 16 Jun 2017 23:52:46 -0700
Subject: [PATCH 2/5] Workaround new warnings generated by GCC 7

src/flash/nor/xmc4xxx.c: In function 'xmc4xxx_get_info_command':
src/flash/nor/xmc4xxx.c:939:43: error: '%d' directive output may be truncated writing between 1 and 10 bytes into a region of size 6 [-Werror=format-truncation=]
     snprintf(otp_str, sizeof(otp_str), "- %d\n", i);
                                           ^~
src/flash/nor/xmc4xxx.c:939:40: note: directive argument in the range [0, 2147483647]
     snprintf(otp_str, sizeof(otp_str), "- %d\n", i);
                                        ^~~~~~~~
src/flash/nor/xmc4xxx.c:939:5: note: 'snprintf' output between 5 and 14 bytes into a destination of size 8
     snprintf(otp_str, sizeof(otp_str), "- %d\n", i);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/flash/nor/xmc4xxx.c | 2 +-
 src/target/arm_adi_v5.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/flash/nor/xmc4xxx.c b/src/flash/nor/xmc4xxx.c
index 02df46a3f..e6c398ff9 100644
--- a/src/flash/nor/xmc4xxx.c
+++ b/src/flash/nor/xmc4xxx.c
@@ -931,7 +931,7 @@ static int xmc4xxx_get_info_command(struct flash_bank *bank, char *buf, int buf_
 
 	/* If OTP Write protection is enabled (User 2), list each
 	 * sector that has it enabled */
-	char otp_str[8];
+	char otp_str[14];
 	if (otp_enabled) {
 		strcat(prot_str, "\nOTP Protection is enabled for sectors:\n");
 		for (int i = 0; i < bank->num_sectors; i++) {
diff --git a/src/target/arm_adi_v5.c b/src/target/arm_adi_v5.c
index d444db2b1..d0d95121f 100644
--- a/src/target/arm_adi_v5.c
+++ b/src/target/arm_adi_v5.c
@@ -1062,7 +1062,7 @@ static int dap_rom_display(struct command_context *cmd_ctx,
 	int retval;
 	uint64_t pid;
 	uint32_t cid;
-	char tabs[7] = "";
+	char tabs[16] = "";
 
 	if (depth > 16) {
 		command_print(cmd_ctx, "\tTables too deep");
-- 
2.13.1

